File: VB/WebSite/App_Code/Person.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Collections.Generic
Imports System.Text


Public Class Person
	Public Const CategorySeparator As String = ", "

	Private id_Renamed As Integer
	Private name_Renamed As String
	Private categories_Renamed As List(Of Category)

	Public Sub New(ByVal id As Integer, ByVal name As String, ParamArray ByVal categories() As Category)
		Me.id_Renamed = id
		Me.name_Renamed = name
		Me.categories_Renamed = New List(Of Category)(categories)
	End Sub

	Public ReadOnly Property ID() As Integer
		Get
			Return id_Renamed
		End Get
	End Property
	Public Property Name() As String
		Get
			Return name_Renamed
		End Get
		Set(ByVal value As String)
			name_Renamed = value
		End Set
	End Property
	Public ReadOnly Property Categories() As List(Of Category)
		Get
			Return categories_Renamed
		End Get
	End Property
	Public ReadOnly Property CategoriesString() As String
		Get
			Dim builder As New StringBuilder()
			For Each item As Category In Categories
				If builder.Length > 0 Then
					builder.Append(CategorySeparator)
				End If
				builder.Append(item.Name)
			Next item
			Return builder.ToString()
		End Get
	End Property
End Class

File: CS/WebSite/App_Code/Person.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Collections.Generic;
using System.Text;

public class Person {
    public const string CategorySeparator = ", ";

    int id;
    string name;
    List<Category> categories;

    public Person(int id, string name, params Category[] categories) {
        this.id = id;
        this.name = name;
        this.categories = new List<Category>(categories);
    }

    public int ID { get { return id; } }
    public string Name {
        get { return name; }
        set { name = value; }
    }
    public List<Category> Categories { get { return categories; } }
    public string CategoriesString {
        get {
            StringBuilder builder = new StringBuilder();
            foreach(Category item in Categories) {
                if(builder.Length > 0)
                    builder.Append(CategorySeparator);
                builder.Append(item.Name);
            }
            return builder.ToString();
        } 
    }
}
